flocker/node/_docker.py

```diff
@@ -242,6 +242,32 @@ def list():
         :return: ``Deferred`` firing with ``set`` of :class:`Unit`.
         """
 
+    def _image_data(image, container, state):
```

Comment:
The interface is for things that are common to all implementations. Private methods are by definition not part of a public interface, and moreover these are needed only due to issues with one implementation; the in-memory implementation doesn't need them.

So, take them off the interface.